### ISSUE TITLE
deal with population overlap

### ISSUE THREAD

**Issue by petrelharp** [bodkan/slendr#72]
We've just realized that since models are WF, and a mate choice callback prohibits mating between individuals not in the overlap between two pops, with high levels of gene flow and low overlap, the individuals in the overlap will be chosen to mate A LOT, leading to bottlenecks in the population. My suggestion for how to fix this is to change the meaning of "gene flow" to be multiplied by the proportion of the population that is overlapped by the other one. This is a breaking change, though!

**Comment by bhaller** [comment]
Or switching to nonWF would semi-automatically fix this problem, since then you could do a search for local mates across multiple subpopulations in whatever way you wanted (i.e., weighted by the "migration rate", while still being realistically constrained to the local area around the choosing individuals).

**Comment by petrelharp** [comment]
True - but I do support @bodkan's goal to get the paper out for v1.0 with WF, I think? Perhaps it's worth having a good brainstorm to make a list of everything that woudl need to be addressed to switch to nonWF. (what does the "gene flow" parameter mean would be one of those...)

**Comment by bhaller** [comment]
> True - but I do support @bodkan's goal to get the paper out for v1.0 with WF, I think?

Indeed.  I'm just saying that rather than trying to fix this problem of overlap/migration in the WF version of slendr, it might be best to just say "this is a problem with WF models, and we will fix it in future by providing a nonWF option, rather than trying to fix it now in the WF context".  Because there's really no end of problems with WF models in a spatial context, and there's no point in trying to band-aid all of them; the right solution, long term, is simply to use nonWF.

**Comment by petrelharp** [comment]
Makes sense. In that case, a strong warning should be added saying "don't have gene flow between barely-overlapping pops"

**Comment by bhaller** [comment]
Certainly not at a significant migration rate!  If m=0.0001 or something then it's maybe OK-ish.  :->
